src/realm/sync/network/network_ssl.hpp

```diff
@@ -1295,20 +1341,34 @@ std::size_t Stream::ssl_perform(Oper oper, std::error_code& ec, Want& want) noex
     std::size_t n;
     std::tie(result, n) = oper();
 
+    // Use caution with MockSSLError, since errSSLWouldBlock will potentially perform
+    // another read that may block
+    if (REALM_UNLIKELY(m_mock_ssl_perform_error && m_mock_ssl_perform_error->operation == m_last_operation)) {
+        result = static_cast<OSStatus>(m_mock_ssl_perform_error->ssl_error);
+        n = static_cast<std::size_t>(m_mock_ssl_perform_error->bytes_processed);
+        if (m_mock_ssl_perform_error->clear_after_access)
+            m_mock_ssl_perform_error.reset();
+    }
+
+    Want blocking_want =
+        m_last_operation ? (*m_last_operation == BlockingOperation::read ? Want::read : Want::write) : Want::nothing;
```

Comment:
I find nested ternary operators a little confusing. You could use a quick lambda to make this a little more readable?
```
Want blocking_want = [&] {
   if (!m_last_operation) {
        return Want::nothing;
   }
   switch(*m_last_operation) {
   case BlockingOperation::read:
       return Want::read;
   case BlockingOperation::write:
       return Want::write;
   }
}();
```